alarm notify auth should use auth-key

From: Ondrej Jirman <ondrej.jirman@zonio.net>

calendar/ChangeLog:

2008-04-06  Ondrej Jirman  <megous@megous.com>

	** Fix for bug #525229

	* gui/alarm-notify/alarm-notify.c: (alarm_notify_add_calendar):

	Use "auth-key" ESource property to determine pass_key if "auth-key" is
	set. This is used by calendar plugins that use common authentication
	credentials for multiple calendars assigned to single calendar account.
---

 calendar/gui/alarm-notify/alarm-notify.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)


diff --git a/calendar/gui/alarm-notify/alarm-notify.c b/calendar/gui/alarm-notify/alarm-notify.c
index 4110681..2a8bacb 100644
--- a/calendar/gui/alarm-notify/alarm-notify.c
+++ b/calendar/gui/alarm-notify/alarm-notify.c
@@ -351,7 +351,9 @@ alarm_notify_add_calendar (AlarmNotify *an, ECalSourceType source_type,  ESource
 	priv = an->priv;
 	str_uri = e_source_get_uri (source);
 	e_uri = e_uri_new (str_uri);
-	if (e_source_get_property (source, "auth-type")) 
+	if (e_source_get_property (source, "auth-key"))
+		pass_key = e_source_get_duped_property (source, "auth-key");
+	else if (e_source_get_property (source, "auth-type")) 
 		pass_key = e_uri_to_string (e_uri, FALSE);
 	else
 		pass_key = g_strdup (str_uri);
